docs/types/value-types.rst

```diff
@@ -663,6 +663,17 @@ Note that public functions of the current contract can be used both as an
 internal and as an external function. To use ``f`` as an internal function,
 just use ``f``, if you want to use its external form, use ``this.f``.
 
+An internal function can be assigned to a variable of an internal function type regardless of where
```

Comment:
Maybe: "A function of internal type" (instead of function of internal visibility)